specs/phase0/fork-choice.md

```diff
@@ -358,8 +362,9 @@ def update_latest_messages(store: Store, attesting_indices: Sequence[ValidatorIn
     target = attestation.data.target
     beacon_block_root = attestation.data.beacon_block_root
     for i in attesting_indices:
-        if i not in store.latest_messages or target.epoch > store.latest_messages[i].epoch:
-            store.latest_messages[i] = LatestMessage(epoch=target.epoch, root=beacon_block_root)
+        if i not in store.has_equivocated:
```

Comment:
suggest prefiltering the set to avoid the extra nested for loop

```python
non_equivocating_attesting_indices = [i for i in attesting_indices if i not in store.has_equivocated
for i in attesting_indices:
    ...